File: VB/Contacts/Controls/IControl.vb
Imports Microsoft.VisualBasic
Imports System

Namespace Contacts.Controls
	Public Enum TabType
		PersonList
		Person
		Preview
		Editor
	End Enum

	Public Interface IControl
		Property Id() As Integer
		ReadOnly Property IsDirty() As Boolean
        Sub Save()
		Sub Reload()
		Function Delete() As Integer
		ReadOnly Property TabType() As TabType
	End Interface
End Namespace

File: CS/Contacts/Controls/IControl.cs
using System;

namespace Contacts.Controls
{
    public enum TabType
    {
        PersonList,
        Person,
        Preview,
        Editor
    }

    public interface IControl
    {
        int Id { get; set; }
        bool IsDirty { get; }
        void Save();
        void Reload();
        int Delete();
        TabType TabType { get; }
    }
}
